pkg/sql/opt/exec/execbuilder/tests/local/local_test: TestExecBuild_select failed

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_select [failed](https://mesolite.cluster.engflow.com/invocations/default/3c1dad29-19c6-479e-93ea-11bfd5034a2a?testReportRun=2&testReportShard=19&testReportAttempt=1#targets-Ly9wa2cvc3FsL29wdC9leGVjL2V4ZWNidWlsZGVyL3Rlc3RzL2xvY2FsOmxvY2FsX3Rlc3Q=) on master @ [5e1ab57f0485ead6fb19461bad9e954a148efe19](https://github.com/cockroachdb/cockroach/commits/5e1ab57f0485ead6fb19461bad9e954a148efe19):


```
[11:17:34] --- progress: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select: 100 statements
[11:17:36] --- progress: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select: 107 statements
[11:17:38] --- progress: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select: 117 statements
[11:17:41] --- progress: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select: 122 statements
[11:17:43] --- progress: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select: 128 statements
[11:17:45] --- progress: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select: 137 statements
[11:17:51] --- progress: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select: 189 statements
[11:17:54] --- progress: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select: 191 statements
[11:17:58] --- progress: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select: 194 statements
[11:18:02] --- progress: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select: 196 statements
[11:18:04] --- progress: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select: 198 statements
[11:18:08] --- progress: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select: 199 statements
    logic.go:3314: let $t_id3 = 127
    logic.go:3264: 
         
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select:1950: SELECT CASE WHEN message LIKE '=== operation:%'
            THEN regexp_replace(
              message,
              ' (gid|node|client|user|txn|cockroach\.\w+|implicit):\S*| hostssl:\S*',
              '', 'g')
            ELSE message
          END
        FROM [SHOW TRACE FOR SESSION]
        WHERE message LIKE 'querying next range at /Table/127/1%' OR
              message LIKE '=== operation:kv.DistSender: sending partial batch%'
        expected:
            querying next range at /Table/127/1/0/0
            querying next range at /Table/127/1/10/0
            querying next range at /Table/127/1/0/0
            querying next range at /Table/127/1/10/0
            querying next range at /Table/127/1/0/0
            === operation:kv.DistSender: sending partial batch _verbose:1
            querying next range at /Table/127/1/10/0
            
        but found (query options: "") :
            querying next range at /Table/127/1/0/0
            querying next range at /Table/127/1/10/0
            querying next range at /Table/127/1/0/0
            querying next range at /Table/127/1/10/0
            querying next range at /Table/127/1/0/0
            === operation:kv.DistSender: sending partial batch _unfinished:1 _verbose:1
            querying next range at /Table/127/1/10/0
            
[11:18:08] --- done: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select with config local: 213 tests, 1 failures
    logic.go:4672: 
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select:1970: error while processing
    logic.go:4672: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select:1970: too many errors encountered, skipping the rest of the input
    panic.go:694: -- test log scope end --
test logs left over in: outputs.zip/logTestExecBuild_select3856815202
--- FAIL: TestExecBuild_select (95.57s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=19</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_select.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63155

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_select [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21326299?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21326299?buildTab=artifacts#/) on master @ [48e86315fbb8a6c898eca0da05b0b5a317783ee2](https://github.com/cockroachdb/cockroach/commits/48e86315fbb8a6c898eca0da05b0b5a317783ee2):


```
=== RUN   TestExecBuild_select
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/a0fd9adea7dd7fca21586817c0273795/logTestExecBuild_select82960990
    test_log_scope.go:82: use -show-logs to present logs inline
    logic.go:3314: let $t_id = 106
    logic.go:3314: let $t_id2 = 107
    logic.go:3314: let $num_ref_bc = 2
    logic.go:3314: let $num_ref_pkey = 3
[18:42:43] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19684/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select: 58 statements
[18:42:46] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19684/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select: 75 statements
[18:42:48] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19684/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select: 87 statements
[18:42:50] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19684/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select: 189 statements
[18:42:52] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19684/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select: 198 statements
    logic.go:3314: let $t_id3 = 127
    logic.go:3264: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19684/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select:1950: SELECT CASE WHEN message LIKE '=== operation:%'
            THEN regexp_replace(
              message,
              ' (gid|node|client|user|txn|cockroach\.\w+|implicit):\S*| hostssl:\S*',
              '', 'g')
            ELSE message
          END
        FROM [SHOW TRACE FOR SESSION]
        WHERE message LIKE 'querying next range at /Table/127/1%' OR
              message LIKE '=== operation:kv.DistSender: sending partial batch%'
        expected:
            querying next range at /Table/127/1/0/0
            querying next range at /Table/127/1/10/0
            querying next range at /Table/127/1/0/0
            querying next range at /Table/127/1/10/0
            querying next range at /Table/127/1/0/0
            === operation:kv.DistSender: sending partial batch _verbose:1
            querying next range at /Table/127/1/10/0
            
        but found (query options: "") :
            querying next range at /Table/127/1/0/0
            querying next range at /Table/127/1/10/0
            querying next range at /Table/127/1/0/0
            querying next range at /Table/127/1/10/0
            querying next range at /Table/127/1/0/0
            === operation:kv.DistSender: sending partial batch _unfinished:1 _verbose:1
            querying next range at /Table/127/1/10/0
            
[18:42:53] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19684/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select with config local: 213 tests, 1 failures
    logic.go:4672: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19684/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select:1970: error while processing
    logic.go:4672: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19684/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/select:1970: too many errors encountered, skipping the rest of the input
    panic.go:694: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a0fd9adea7dd7fca21586817c0273795/logTestExecBuild_select82960990
--- FAIL: TestExecBuild_select (12.38s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_select.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/opt/exec/execbuilder/testdata/select
+++ b/pkg/sql/opt/exec/execbuilder/testdata/select
@@ -25,7 +25,7 @@ SELECT
   CASE WHEN message LIKE '=== operation:%'
     THEN regexp_replace(
       regexp_replace(message, ' statement:.*?( _| $)', '\1'),
-      ' (gid|node|client|user|txn|cockroach\.\w+|implicit):\S*| hostssl:\S*',
+      ' (gid|node|client|user|txn|cockroach\.\w+|implicit|_unfinished):\S*| hostssl:\S*',
       '', 'g')
     ELSE regexp_replace(message, 'pos:[0-9]*', 'pos:?')
   END,
@@ -33,7 +33,7 @@ SELECT
 FROM [SHOW TRACE FOR SESSION]
 WHERE message LIKE '%=== operation:%' OR message LIKE '%pos%executing%';
 ----
-0   === operation:session recording _unfinished:1 _verbose:1  session recording
+0   === operation:session recording _verbose:1                session recording
 1   === operation:session tracing _verbose:1                  session tracing
 1   [Open pos:?] executing Sync                               session tracing
 2   === operation:commit sql txn _verbose:1                   commit sql txn
@@ -1951,7 +1951,7 @@ query T
 SELECT CASE WHEN message LIKE '=== operation:%'
     THEN regexp_replace(
       message,
-      ' (gid|node|client|user|txn|cockroach\.\w+|implicit):\S*| hostssl:\S*',
+      ' (gid|node|client|user|txn|cockroach\.\w+|implicit|_unfinished):\S*| hostssl:\S*',
       '', 'g')
     ELSE message
   END

--- a/pkg/sql/opt/exec/execbuilder/testdata/vectorize_local
+++ b/pkg/sql/opt/exec/execbuilder/testdata/vectorize_local
@@ -258,7 +258,7 @@ query T rowsort
 SELECT DISTINCT CASE WHEN message LIKE '=== operation:%'
     THEN regexp_replace(
       message,
-      ' (gid|node|client|user|txn|cockroach\.\w+|implicit):\S*| hostssl:\S*',
+      ' (gid|node|client|user|txn|cockroach\.\w+|implicit|_unfinished):\S*| hostssl:\S*',
       '', 'g')
     ELSE message
   END

